aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp

```diff
@@ -1042,23 +1042,37 @@ Tensor reduce_sparse_csr_dim0_cpu_template(const Tensor& sparse, ReductionOp rop
   new_crow_indices[1] = nnz;
 
   Tensor new_values = at::empty({nnz}, values.options());
-  new_values.fill_(rop.identity());
+  bool need_acc = (values.scalar_type() == kHalf || values.scalar_type() == kBFloat16 || values.scalar_type() == kComplexHalf);
```

Comment:
Just a note: in principle, the `need_acc` information is defined by `scalar_t` and `ReductionOp`. This would allow compile-time optimizations because the if-blocks on `need_acc` can use `constexpr`. For instance, if `ReductionOp` would store its template typename as a `type` member, we could have:
```
constexpr bool need_acc = !std::is_same<scalar_t, rop::type>::value;
```